presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -62,14 +62,17 @@
     // for each source, the list of inputs corresponding to each output
     private final List<List<Symbol>> inputs;
 
+    private final boolean orderSensitive;
+
     @JsonCreator
     public ExchangeNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("type") Type type,
             @JsonProperty("scope") Scope scope,
             @JsonProperty("partitioningScheme") PartitioningScheme partitioningScheme,
             @JsonProperty("sources") List<PlanNode> sources,
-            @JsonProperty("inputs") List<List<Symbol>> inputs)
+            @JsonProperty("inputs") List<List<Symbol>> inputs,
+            @JsonProperty("orderSensitive") boolean orderSensitive)
```

Comment:
IMO `ExhcangeNode` should always be order sensitive, order of rows should be maintained. A lot of optimization depends on that, so I would make `ExchangeNode` to preserve order always.